rust/processor/src/models/token_v2_models/v2_token_activities.rs

```diff
@@ -158,11 +158,19 @@ impl TokenActivityV2 {
             // burn and mint events are attached to the collection. The rest should be attached to the token
             let token_data_id = match token_event {
                 V2TokenEvent::MintEvent(inner) => inner.get_token_address(),
+                V2TokenEvent::ConcurrentMintEvent(inner) => inner.get_token_address(),
                 V2TokenEvent::BurnEvent(inner) => inner.get_token_address(),
+                V2TokenEvent::ConcurrentBurnEvent(inner) => inner.get_token_address(),
                 V2TokenEvent::TransferEvent(inner) => inner.get_object_address(),
                 _ => event_account_address.clone(),
             };
 
+            let event_type_override = match event_type.as_str() {
+                "0x4::collection::ConcurrentMintEvent" => "0x4::collection::MintEvent".to_string(),
+                "0x4::collection::ConcurrentBurnEvent" => "0x4::collection::BurnEvent".to_string(),
+                _ => event_type.clone(),
+            };
```

Comment:
yup this is what we want but may I suggest merging this with the match block below.

Reply:
Done.